src/clarity-angular/select/select-menu-trigger.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2016 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {Directive, ElementRef, EventEmitter, HostListener, Input, OnChanges, Output} from "@angular/core";
+import {Observable} from "rxjs/Observable";
+
+import {IfOpenService} from "../utils/conditional/if-open.service";
+
+import {RootSelectService} from "./providers/select.service";
+import {Select} from "./select";
+
+
+@Directive({selector: "[clrSelectMenuTrigger]"})
```

Comment:
Is this directive needed? Instead of `<clr-icon aria-hidden="true" shape="angle down" clrSelectMenuTrigger></clr-icon>`, if you wrap the `clr-icon` with a button and attach a click event on it, the directive can be completely removed, no?

Reply:
Agreed.